### PR TITLE
Added terrible implementation of basic dashboard to update links without...

### PR DESCRIPTION
**Pull request by spencerldixon**
- Added admin::dashboard controller for building an admin dashboard
- Added edit action to short urls
- Added admin boolean to users table
- Added links for admin pages
- Added dedicated admin layout
- Added datatable in admin index view for searching for links

### CODE REVIEW

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -1,3 +1,8 @@
 class ApplicationController < ActionController::Base
   protect_from_forgery
+
+  protected
+    def authenticate_admin
```
Comment (lucidstack) [inline app/controllers/application_controller.rb]: Why the indentation here? Also, trying the new integration with Slack! :grinning:

Comment (spencerldixon) [inline app/controllers/application_controller.rb]: No idea! I've always indented protected and private methods as an aesthetic way of identifying them in a class. Pretty sure I picked it up from a ruby book somewhere

File: Gemfile [hunk 2]
```diff
@@ -25,7 +25,7 @@ group :development do
   gem 'guard-minitest'
   gem 'rb-fsevent', '~> 0.9'
 
-  gem 'debugger'
```
Comment (agnessa) [inline Gemfile]: debugger does not support ruby 2.0, byebug is recommended

**Comment by agnessa** [comment]
you'll need to add admin.js to list of precompiled assets possibly

**Comment by agnessa** [comment]
The datatables js / css should go in vendor/assets and not app/assets

**Pull request closed without merging**